lib/array-slice.js

```diff
@@ -19,24 +23,45 @@ var ArraySlice = createClass({
 
   // High Order Functions
 
+  /**
+   * @param {callback} - Function to execute for each element
+   * @param {thisArg} - Value to use as this (i.e the reference Object) when executing callback
+   * @memberof ArraySlice.prototype
```

Comment:
Since we use ES5 which doesn't have classes, we have to add additional annotations like `@memberof`, `@readonly` etc. If we move to ES6 eventually (I could see this happening), we get to remove some of these as they can be inferred.